cloud/amazon/cloudformation_facts.py

```diff
@@ -0,0 +1,267 @@
+
+    def paginated_response(self, func, result_key, next_token=None):
+        '''
+        Returns expanded response for paginated operations.
+        The 'result_key' is used to define the concatenated results that are combined from each paginated response.
+        '''
+        args=dict()
+        if next_token:
+            args['NextToken'] = next_token
+        response = func(**args)
+        result = response.get(result_key)
+        next_token = response.get('NextToken')
+        if not next_token:
+           return result
+        return result + self.paginated_response(func, result_key, next_token)
+
+def to_dict(items, key, value):
+    ''' Transforms a list of items to a Key/Value dictionary '''
+    if items:
+        return dict(zip([i[key] for i in items], [i[value] for i in items]))
+    else:
+        return dict()
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        stack_name=dict(required=True, type='str' ),
+        all_facts=dict(required=False, default=False, type='bool'), 
+        stack_policy=dict(required=False, default=False, type='bool'),
+        stack_events=dict(required=False, default=False, type='bool'),
+        stack_resources=dict(required=False, default=False, type='bool'),
+        stack_template=dict(required=False, default=False, type='bool'),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+
+    if not HAS_BOTO3:
+      module.fail_json(msg='boto3 is required.')
+    
+    # Describe the stack            
+    service_mgr = CloudFormationServiceManager(module)
+    stack_name = module.params.get('stack_name')
+    result = dict()
+    result['stack_description'] = service_mgr.describe_stack(stack_name)
+
+    # Create stack output and stack parameter dictionaries
+    if result['stack_description']:
+        result['stack_outputs'] = to_dict(result['stack_description'].get('Outputs'), 'OutputKey', 'OutputValue')    
+        result['stack_parameters'] = to_dict(result['stack_description'].get('Parameters'), 'ParameterKey', 'ParameterValue')    
+
+    # Create optional stack outputs
+    all_facts = module.params.get('all_facts')
+    if all_facts or module.params.get('stack_resources'):
+        result['stack_resource_list'] = service_mgr.list_stack_resources(stack_name)
+        result['stack_resources'] = to_dict(result.get('stack_resource_list'), 'LogicalResourceId', 'PhysicalResourceId')
+    if all_facts or module.params.get('stack_template'):
+        result['stack_template'] = service_mgr.get_template(stack_name)
+    if all_facts or module.params.get('stack_policy'):
+        result['stack_policy'] = service_mgr.get_stack_policy(stack_name)
+    if all_facts or module.params.get('stack_events'):
+        result['stack_events'] = service_mgr.describe_stack_events(stack_name)
+
+    result['changed'] = False
+    module.exit_json(**result)
```

Comment:
it doesn't look like this module is actually returning facts.  You have to use a special dictionary key for the results so that the controller-side code knows to turn them into facts: http://docs.ansible.com/ansible/developing_modules.html#module-provided-facts